src/FixtureBuilder/Denormalizer/Fixture/Chainable/NullRangeNameReferenceDenormalizer.php

```diff
@@ -0,0 +1,193 @@
+    {
+        return 1 === preg_match(self::REGEX, $name, $matches);
+    }
+
+    /**
+     * @inheritdoc
+     */
+    public function denormalize(
+        FixtureBag $builtFixtures,
+        string $className,
+        string $fixtureId,
+        array $specs,
+        FlagBag $flags
+    ): FixtureBag {
+        $matches = [];
+        if (false === $this->canDenormalize($fixtureId, $matches)) {
+            throw LogicExceptionFactory::createForCannotDenormalizerForChainableFixtureBuilderDenormalizer(__METHOD__);
+        }
+
+        $referencedName = $matches['name'];
+        $allFlag = ($matches['flag'] ?? null) === '*';
+
+        $fixtureIds = $this->buildReferencedValues($builtFixtures, $referencedName, $allFlag);
+
+        $fixtureIdPrefix = $this->determineFixtureIdPrefix($fixtureId);
+
+        foreach ($fixtureIds as $referencedFixtureId => $valueForCurrent) {
+            $builtFixtures = $builtFixtures->with(
+                $this->buildFixture(
+                    $fixtureIdPrefix . $referencedFixtureId,
+                    $className,
+                    $specs,
+                    $flags,
+                    $valueForCurrent
+                )
+            );
+        }
+
+        return $builtFixtures;
+    }
+
+    /**
+     * @param FixtureBag $builtFixtures
+     * @param string     $referencedName
+     * @param bool       $allFlag
+     *
+     * @return array
+     */
+    private function buildReferencedValues(
+        FixtureBag $builtFixtures,
+        string $referencedName,
+        bool $allFlag
+    ): array {
+        if (false === $allFlag) {
+            $fixture = $builtFixtures->get($referencedName);
+
+            return [$referencedName => $fixture];
+        }
+
+        $matchedFixtures = array_filter(
+            $builtFixtures->toArray(),
+            function (string $referenceName) use ($referencedName) {
+                return strpos($referenceName, $referencedName) === 0;
+            },
+            ARRAY_FILTER_USE_KEY
+        );
+
+        return $matchedFixtures;
+    }
+
+    /**
+     * @param string $fixtureId
+     *
+     * @return string
+     * @throws \LogicException
+     */
+    private function determineFixtureIdPrefix(
+        string $fixtureId
+    ): string {
+        $matches = [];
+        if (false === $this->canDenormalize($fixtureId, $matches)) {
+            throw LogicExceptionFactory::createForCannotDenormalizerForChainableFixtureBuilderDenormalizer(__METHOD__);
+        }
+
+        return str_replace(
+            sprintf('{%s}', $matches['expression']),
+            '',
+            $matches[0]
+        );
+    }
+
+    /**
+     * @param string           $fixtureId
+     * @param string           $className
+     * @param array            $specs
+     * @param FlagBag          $flags
+     * @param FixtureInterface $valueForCurrent
+     *
+     * @return FixtureInterface
+     * @throws \Nelmio\Alice\Throwable\DenormalizationThrowable
```

Comment:
this class can be imported; since the other phpdoc elements are redundant we can get rid of the others